api/agent/slots.go

```diff
@@ -352,6 +352,13 @@ func getSlotQueueKey(call *call) string {
 		hash.Write(unsafeBytes("\x00"))
 	}
 
+	for k,v := range call.extensions {
```

Comment:
additionally to the prior comment (agreed), this won't work as intended either because map iteration is random, meaning these will have to get sorted (see annotations block above this) or new containers will get launched for no real reason (it is sad, if we had func versioning it may not be needed and we could save some bytes too). it's unclear from context on the PR which specific keys are of interest here, too

Reply:
Thanks for the review. Updated the change

Reply:
To address the question of context: the original problem observed was that if an auth token passed through in an extension expired before a successful pull could occur for a slot, all subsequent pulls would fail.  There are other things like tokens that get passed through to containers in some cases.  The original concern was that expiry of these kinds of objects would also cause similar problems, but after consulting with those authors it looks like it's just the auth token.  We could potentially simplify here to simply hash the output of the auth token's function, which already allows plug-able implementations.  That said, the next extension that impacted container behavior would need to implement this, so having something more generic earlier may be a good thing.

Reply:
This was changed to use a hash of the extensions